src/test/java/BaseTest.java

```diff
@@ -1,11 +1,51 @@
+import Pages.*;
 import io.github.bonigarcia.wdm.WebDriverManager;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.chrome.ChromeDriver;
 import org.openqa.selenium.chrome.ChromeOptions;
+import org.openqa.selenium.interactions.Actions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
 import org.testng.annotations.BeforeSuite;
-
+import java.time.Duration;
 public class BaseTest {
-
+    public WebDriver driver;
+    public Actions actions;
+    public String url = "https://qa.koel.app/";
+    WebDriverWait wait;
+    BasePage basepage;
+    LoginPage loginPage;
+    HomePage homePage;
+    ProfilePage profilePage;
+    RegistrationPage registrationPage;
+    SearchPage searchPage;
+    PlaylistPage playlistPage;
     @BeforeSuite
     static void setupClass() {
         WebDriverManager.chromedriver().setup();
     }
-}
\ No newline at end of file
+    @BeforeMethod
+    public void launchBrowser() {
+        ChromeOptions options = new ChromeOptions();
+        options.addArguments("--remote-allow-origins=*");
+        options.addArguments("--disable-notifications");
+        options.addArguments("--start-maximized");
+        driver = new ChromeDriver(options);
+        wait = new WebDriverWait(driver,Duration.ofSeconds(10));
+        actions = new Actions(driver);
+        basepage = new BasePage(driver, wait, actions);
+        basepage.openLoginUrl(url);
+        loginPage = new LoginPage(driver, wait, actions);
+        homePage = new HomePage(driver, wait, actions);
+        profilePage = new ProfilePage(driver, wait, actions);
+        registrationPage = new RegistrationPage(driver, wait, actions);
+        searchPage = new SearchPage(driver, wait, actions);
+        playlistPage = new PlaylistPage(driver, wait, actions);
+//        driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(20));
+    }
+    @AfterMethod
+    public void closeBrowser() {
+        basepage.quitBrowser();
+    }
+}
```

Comment:
**HINT**

1. **BasePage and BaseTest** 

**BaseTest** — Implements the necessary methods for work with webdriver.

The BaseTest class contains common setup and teardown methods, as well as any shared functionality or configurations required by multiple test classes.

**BaseTest**

The problem we are trying to solve here is the reuse of common behaviors across test classes avoiding coding duplication and centralizing these actions at one point.

**@BeforeSuite**	- The annotated method will be run before all tests in this suite have run.

**@BeforeMethod** -	The annotated method will be run before all the test methods in the current class have been run.

**@AfterMethod**	- The annotated method will be run after each test method.

_Example without a Base Test Class_

![image](https://user-images.githubusercontent.com/104761482/233972627-2d7ffdc7-6a2f-4e9f-8a3a-a1b4a7a47bc4.png)


**BasePage**  — class serves as a foundation or parent class for individual page classes. It contains common methods and functionalities that are shared across multiple pages in an application.

_Example without a Base Page Class_

![image](https://user-images.githubusercontent.com/104761482/236891493-7f7013c7-fde2-4bf9-9084-3aa7483df8e3.png)


2. **Test in Page Object Model (POM)**

![Clean_Test](https://user-images.githubusercontent.com/104761482/236863813-f30d7bfc-e89d-4419-bdea-4b1517155e39.png)